### PR TITLE
Doc comments for src/arch/mod.rs

### PR DESCRIPTION
**Pull request by pythondude325**
I realized you made this module private, but I already wrote the comments, so I will push them.

### CODE REVIEW

File: src/arch/mod.rs [hunk 1]
```diff
@@ -17,14 +19,18 @@ use crate::data::{
 };
 use Type::*;
 
-// NOTE: this is required by the standard to always be one
+/// Size of a `char` in bytes;
```
Comment (jyn514) [inline src/arch/mod.rs]:
```suggestion
/// Size of a `char` in bytes
```

File: src/arch/mod.rs [hunk 2]
```diff
@@ -17,14 +19,18 @@ use crate::data::{
...
+///
+/// Note: The standard requires this to always be one.
```
Comment (jyn514) [inline src/arch/mod.rs]:
```suggestion
/// Note: The standard requires this to always be one.
/// http://port70.net/~nsz/c/c11/n1570.html#6.5.3.4
```

File: src/arch/mod.rs [hunk 3]
```diff
@@ -17,14 +19,18 @@ use crate::data::{
...
 const CHAR_SIZE: u16 = 1;
 
 // TODO: allow this to be configured at runtime
+/// The target triple for the configured target.
```
Comment (jyn514) [inline src/arch/mod.rs]: I don't think this is very helpful. Maybe link to an explanation of target triple instead: https://clang.llvm.org/docs/CrossCompilation.html#target-triple or https://docs.rs/target-lexicon/0.10.0/target_lexicon/struct.Triple.html

Also, this is the host target, not any configured target.

File: src/arch/mod.rs [hunk 4]
```diff
@@ -83,6 +89,7 @@ impl StructType {
 }
 
 impl Type {
+    /// Returns true if `other` can be cast to `self` without losing infomation.
```
Comment (jyn514) [inline src/arch/mod.rs]:
```suggestion
    /// Returns true if `other` can be converted to `self` without losing infomation.
```

Cast makes me think an explicit cast (`(int)1.4`), I think conversion is better.

File: src/arch/mod.rs [hunk 5]
```diff
@@ -149,9 +160,11 @@ impl Type {
             Error => Err("cannot take `alignof` <type error>"),
         }
     }
+    /// Get the pointer type
```
Comment (jyn514) [inline src/arch/mod.rs]: This is not really an accurate description. This returns an integer type which is large enough to represent the pointer; this is an IR-level type and has no relation to any C type.

File: src/arch/mod.rs [hunk 6]
```diff
@@ -149,9 +160,11 @@ impl Type {
...
     pub fn ptr_type() -> IrType {
         IrType::int(CHAR_BIT * PTR_SIZE).expect("pointer size should be valid")
     }
+    /// Convert `self` into the type for the IR.
```
Comment (jyn514) [inline src/arch/mod.rs]:
```suggestion
    /// Return an IR type which can represent this C type
```

'Convert' suggests that it consumes `self`.

File: src/arch/mod.rs (hunk 3 above)
Comment (pythondude325) [reply src/arch/mod.rs]: My new explanation for target triple may be a little long winded.

Comment (jyn514) [reply src/arch/mod.rs]: I like it :)

**Pull request merged**